src/customer/strategies/apple-pay/apple-pay-customer-strategy.ts

```diff
@@ -0,0 +1,377 @@
+
+        this._applePayButton = this._createButton(container);
+        this._applePayButton.addEventListener('click', this._handleWalletButtonClick);
+
+        return this._store.getState();
+    }
+
+    deinitialize(): Promise<InternalCheckoutSelectors> {
+        return Promise.resolve(this._store.getState());
+    }
+
+    signIn(): Promise<InternalCheckoutSelectors> {
+        throw new NotImplementedError(
+            'In order to sign in via Apple, the shopper must click on "Apple Pay" button.'
+        );
+    }
+
+    signOut(): Promise<InternalCheckoutSelectors> {
+        throw new NotImplementedError(
+            'Need to do signout via apple.'
+        );
+    }
+
+    executePaymentMethodCheckout(options?: ExecutePaymentMethodCheckoutOptions): Promise<InternalCheckoutSelectors> {
+        options?.continueWithCheckoutCallback?.();
+
+        return Promise.resolve(this._store.getState());
+    }
+
+    private _createButton(containerId: string): HTMLElement {
+        const container = document.getElementById(containerId);
+
+        if (!container) {
+            throw new InvalidArgumentError('Unable to create sign-in button without valid container ID.');
+        }
+
+        const button = document.createElement('input');
+        button.type = 'image';
+        button.src = this._paymentMethod?.logoUrl || '';
+        button.style.height = '40px';
+        button.style.width = '75px';
+        container.appendChild(button);
+
+        return button;
+    }
+
+    @bind
+    private _handleWalletButtonClick(event: Event) {
+        event.preventDefault();
+        const state = this._store.getState();
+        const cart = state.cart.getCartOrThrow();
+        const config = state.config.getStoreConfigOrThrow();
+        if (!this._paymentMethod || !this._paymentMethod.initializationData) {
+            throw new MissingDataError(MissingDataErrorType.MissingPaymentMethod);
+        }
+        const request = this._getBaseRequest(cart, config, this._paymentMethod);
+        const applePaySession = this._sessionFactory.create(request);
+        this._handleApplePayEvents(applePaySession, this._paymentMethod, config);
+
+        applePaySession.begin();
+    }
+
+    private _getBaseRequest(
+        cart: Cart,
+        config: StoreConfig,
+        paymentMethod: PaymentMethod
+    ): ApplePayJS.ApplePayPaymentRequest {
+        const { storeProfile: { storeCountryCode, storeName } } = config;
+        const { currency: { code } } = cart;
+
+        const { initializationData : { merchantCapabilities, supportedNetworks } } = paymentMethod;
+
+        return {
+            requiredBillingContactFields: ['postalAddress'],
+            requiredShippingContactFields: ['email', 'phone', 'postalAddress'],
+            countryCode: storeCountryCode,
+            currencyCode: code,
+            merchantCapabilities,
+            supportedNetworks,
+            lineItems: [],
```

Comment:
Have you tested the flow where a shopper has filled in everything, and then goes back to the customer step and click on the ApplePay button. Would they be forced to start from the beginning?

Reply:
This is one of the unresolved questions as to what we do when we close the apple pay modal. I did want to check with you that should we reset the checkout at this instance?

Reply:
What you described is a different flow to what I described. But yes, it's another flow we need to consider. I suppose it does make sense to reset the checkout details.

Reply:
Will probably sync on this once we are back from holidays.

Reply:
In my dev store, when I goes back to the customer step by clicking `sign out`. `The shipping price you were quoted is no longer valid. Click OK to see the most up-to-date shipping prices` pops up. Then, everything reset.

https://user-images.githubusercontent.com/88361607/147194541-cbc0cc1f-48fe-4047-9c62-3e7ca5f6ac10.mov